app/components/signature-pill.js

```diff
@@ -11,84 +11,49 @@ export default class SignaturePillComponent extends Component {
   @service intl;
   @service currentSession;
 
-  @tracked signingHubUrl;
-  @tracked isMarked = false;
-  @tracked isPrepared = false;
-  @tracked hasToBeApproved = false;
-  @tracked hasToBeSigned = false;
-  @tracked isSigned = false;
-  @tracked isRefused = false;
-  @tracked isCancelled = false;
-
-  constructor() {
-    super(...arguments);
-
-    this.loadData.perform();
-  }
+  status = trackedFunction(this, async () => {
+    const signMarkingActivity = await this.args.signMarkingActivity;
+    const signSubcase = await signMarkingActivity.signSubcase;
 
-  get skin() {
-    if (this.isRefused || this.isCancelled) {
-      return "error";
-    } else if (this.signingHubUrl) {
-      return "link"
-    } else {
-      return "ongoing"
-    }
-  }
+    return await (await signSubcase.signFlow).status;
```

Comment:
I get why we need to `await` both here, but I get shivers up and down my spine when I see this double `await` 😅

I might put the signflow in a variable here

Reply:
In hindsight. Couldn't this just be an async getter?

I suppose that would clutter the rest of the code with `await this.status` too much. I'm ok with it for now, but would prefer someone who works in Ember more often than me to take a look at it.

Reply:
What do you mean by async getter? Because as far as I know there is no such thing